sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/ChangeFeedBatch.scala

```diff
@@ -19,10 +20,12 @@ private class ChangeFeedBatch
 ) extends Batch
   with CosmosLoggingTrait {
 
+  val batchId = UUID.randomUUID().toString()
   logTrace(s"Instantiated ${this.getClass.getSimpleName}")
 
   override def planInputPartitions(): Array[InputPartition] = {
 
+    logInfo(s"--> planInputPartitions $batchId")
```

Comment:
nit: Should this be debug or other level? The text doesn't seem to be something the user can make sense of?

Reply:
I thought about it. The main reason I made this info is that I want preview customers to always be able to see the log (and Databricks default level is Info). I think we need to make a bigger change with regards to logging all-up - I have some ideas, but this will be a task at least for a week (we need correlated logs with 5-min aggregations of metrics that we can expose by default - so to avoid noise but still get meaningful info by default) Similar to some of the work Naveen did in Java - just for the purpose of Spark,.